tests/machine_control/test_machine_controller.py

```diff
@@ -718,6 +718,46 @@ def test_read_struct_field(self, x, y, p, which_struct, field, expected):
             x, y, p
         )
 
+    @pytest.mark.parametrize("x, y, p", [(0, 1, 2), (2, 5, 6)])
+    @pytest.mark.parametrize(
+        "which_struct, field, value",
+        [("sv", "dbg_addr", 5),
+         ("sv", "status_map", (0, )*20),
+         ])
+    def test_write_struct_field(self, x, y, p, which_struct, field, value):
+        # Open the struct file
+        struct_data = pkg_resources.resource_string("rig", "boot/sark.struct")
+        structs = struct_file.read_struct_file(struct_data)
+        assert (six.b(which_struct) in structs and
+                six.b(field) in structs[six.b(which_struct)]), "Test is broken"
+
+        # Create the mock controller
+        cn = MachineController("localhost")
```

Comment:
No live hardware test? Probably a good idea! `sv->led_period` can be safely written to 0 and then written back to its original value. (This enables/disables the flashing LEDs on the board).